### ISSUE TITLE
Warning: QgsVectorFileWriter.writeAsVectorFormat() is deprecated

### ISSUE THREAD

**Issue by agiudiceandrea** [qgis2web/qgis2web#1056]
QGIS 3.34.5 / qgis2web 3.19.0 / Windows 10

```
 WARNING    warning:C:\Users/Andrea/AppData/Roaming/QGIS/QGIS3\profiles\qgis334_7/python/plugins\qgis2web\utils.py:254: DeprecationWarning: QgsVectorFileWriter.writeAsVectorFormat() is deprecated
              e, err = QgsVectorFileWriter.writeAsVectorFormat(cleanLayer, tmpPath,
             
             traceback: File "C:\Users/Andrea/AppData/Roaming/QGIS/QGIS3\profiles\qgis334_7/python/plugins\qgis2web\qgis2web.py", line 59, in run
              self.dlg = MainDialog(self.iface)
              File "C:\Users/Andrea/AppData/Roaming/QGIS/QGIS3\profiles\qgis334_7/python/plugins\qgis2web\maindialog.py", line 179, in __init__
              self.autoUpdatePreview()
              File "C:\Users/Andrea/AppData/Roaming/QGIS/QGIS3\profiles\qgis334_7/python/plugins\qgis2web\maindialog.py", line 369, in autoUpdatePreview
              self.previewMap()
              File "C:\Users/Andrea/AppData/Roaming/QGIS/QGIS3\profiles\qgis334_7/python/plugins\qgis2web\maindialog.py", line 372, in previewMap
              preview_file = self.createPreview()
              File "C:\Users/Andrea/AppData/Roaming/QGIS/QGIS3\profiles\qgis334_7/python/plugins\qgis2web\maindialog.py", line 334, in createPreview
              return writer.write(self.iface,
              File "C:\Users/Andrea/AppData/Roaming/QGIS/QGIS3\profiles\qgis334_7/python/plugins\qgis2web\olwriter.py", line 81, in write
              self.preview_file = self.writeOL(iface, feedback,
              File "C:\Users/Andrea/AppData/Roaming/QGIS/QGIS3\profiles\qgis334_7/python/plugins\qgis2web\olwriter.py", line 130, in writeOL
              exportLayers(iface, layers, folder, precision, optimize,
              File "C:\Users/Andrea/AppData/Roaming/QGIS/QGIS3\profiles\qgis334_7/python/plugins\qgis2web\utils.py", line 232, in exportLayers
              exportVector(layer, sln, layersFolder, restrictToExtent,
              File "C:\Users/Andrea/AppData/Roaming/QGIS/QGIS3\profiles\qgis334_7/python/plugins\qgis2web\utils.py", line 254, in exportVector
              e, err = QgsVectorFileWriter.writeAsVectorFormat(cleanLayer, tmpPath,
             
```

**Comment by andreaordonselli** [comment]
thanks, can you provide me with an example file to test with and which now gives the error?

**Comment by agiudiceandrea** [comment]
@andreaordonselli, the function writeAsVectorFormat is deprecated since QGIS 3.10. To avoid the warning message, the writeAsVectorFormatV2 function should be used since QGIS 3.10 and the writeAsVectorFormatV3 function should be used since QGIS 3.20.
https://qgis.org/pyqgis/master/core/QgsVectorFileWriter.html#qgis.core.QgsVectorFileWriter.writeAsVectorFormat

To spot the warning, just use a project with the World Map layer (type the word "world" in the map coordinate widget) and create an OpenLayer web map using default settings.

**Comment by andreaordonselli** [comment]
I have just released version [3.20.0](https://github.com/qgis2web/qgis2web/releases/tag/v3.20.0), which fixes the issues discussed in this thread. Shortly, the update will be available within the QGIS plugin repository.

Thank you for using the plugin.

Best regards.
